worlds/pokemon_rb/regions.py

```diff
@@ -1609,19 +1613,19 @@ def create_regions(self):
     connect(multiworld, player, "Menu", "Pokedex", one_way=True)
     connect(multiworld, player, "Menu", "Evolution", one_way=True)
     connect(multiworld, player, "Menu", "Fossil", lambda state: logic.fossil_checks(state,
-        state.multiworld.second_fossil_check_condition[player].value, player), one_way=True)
+        state.multiworld.worlds[player].options.second_fossil_check_condition.value, player), one_way=True)
     connect(multiworld, player, "Pallet Town", "Route 1")
     connect(multiworld, player, "Route 1", "Viridian City")
     connect(multiworld, player, "Viridian City", "Route 22")
     connect(multiworld, player, "Route 22", "Route 22-F", lambda state: state.has("Defeat Viridian Gym Giovanni", player), one_way=True)
     connect(multiworld, player, "Route 2-SW", "Route 2-Grass", one_way=True)
     connect(multiworld, player, "Route 2-NW", "Route 2-Grass", one_way=True)
     connect(multiworld, player, "Route 22 Gate-S", "Route 22 Gate-N",
-            lambda state: logic.has_badges(state, state.multiworld.route_22_gate_condition[player].value, player))
-    connect(multiworld, player, "Route 23-Grass", "Route 23-C", lambda state: logic.has_badges(state, state.multiworld.victory_road_condition[player].value, player))
+            lambda state: logic.has_badges(state, state.multiworld.worlds[player].options.route_22_gate_condition.value, player))
```

Comment:
You have a world instance here, why do you need to go through state then multiworld then player?

Reply:
Changed.